apiv1/controllers/SceneController.js

```diff
@@ -337,5 +337,88 @@ module.exports = {
             id: scene._id,
             imgError: imgError
         });
+    },
+    export: async function(req, resp) {
+        const exportFields = {
+            name: 1,
+            settings: 1,
+            code: 1,
+            desc: 1,
+            _id: 1
+        };
+        let id = req.query.id;
+        if(!req.headers['x-access-token']) {
+            return resp.status(400).json({
+                message: "Missing user ID",
+                error: "Bad Request"
+            });
+        }
+
+        let uid = await verifyGoogleToken(req.headers['x-access-token']);
+        if(!uid) {
+            return resp.status(401).json(invalid_token);
+        }
+
+        let scenes = [];
+        //Export all scenes if no id is specified in the query
+        if(!id) {
+            scenes = await SceneSchema.find({"uid": uid}, exportFields);
+        } else {
+            scenes = await SceneSchema.find({"uid": uid, "_id": ObjectId(id)}, exportFields);
+        }
+        
+        if(scenes.length <= 0) {
+            return resp.status(204).send();
+        }
+
+        let output = [];
+        scenes.forEach((scene) => {
+            const imgBase64 = fs.readFileSync(`${destFolder}/${scene._id}.jpg`).toString('base64');
+            const sceneObj = scene.toObject();
+
+            sceneObj.image = imgBase64;
+            output.push(sceneObj);
+        });
+
+        return resp.status(200).json(output);
+    },
+    import: async function(req, resp) {
+        let body = req.body;
+        if(!req.headers['x-access-token']) {
+            return resp.status(400).json({
+                message: "Missing user ID",
+                error: "Bad Request"
+            });
+        }
+
+        let importedScenes = [];
+        let uid = await verifyGoogleToken(req.headers['x-access-token']);
+        if(!uid) {
+            return resp.status(401).json(invalid_token);
+        }
+
+        for(let scene of body) {
```

Comment:
Is there a limit of how many scenes we can import per json file? If there's like a lot of scenes to import without any limit, would that be a problem?

Reply:
Currently there is no limit, since this happens asynchronously shouldn't cause massive issues, the only thing that could happen is someone posting large JSON files, which could cause long processing times.  

The backend is threaded process for requests, so it shouldn't be that much of a problem.